pkg/keystore/keystore.go

```diff
@@ -123,7 +123,10 @@ func (ks *Keystore) MaskTrustedKeySystemPrefix(prefix, fingerprint string) (stri
 		return "", err
 	}
 	dst := path.Join(ks.LocalPrefixPath, acidentifier.String(), fingerprint)
-	return dst, ioutil.WriteFile(dst, []byte(""), 0644)
+	if err := ioutil.WriteFile(dst, []byte(""), 0644); err != nil {
+		return dst, err
+	}
+	return dst, os.Chmod(dst, 0644)
```

Comment:
I thought that ```WriteFile``` changes the permissions of the created file, there I'm wondering what is the need of making an additional call?

Reply:
Not if the system has strict umask set. Check #3779 
With umask set to 022 everything is fine.

Reply:
@ybubnov it does, however `perm` here still has to go through umask masking, so the result may be different than `0644`. (I didn't check for `WriteFile` explicitly, but other golang stdlib functions have this behavior wrt. perms)